src/box/lua/schema.lua

```diff
@@ -2508,6 +2508,8 @@ local function check_select_safety(index, key_is_nil, itype, limit, offset,
     end
 end
 
+box.internal.check_select_safety = check_select_safety -- for read view
```

Comment:
This function can't be used for read views as is, because it uses `box.space` to retrieve the space name, which may be unavailable at the time a read view is used (a space may be dropped after read view creation).

Also, an index read view object doesn't have `space_id` field, although we could probably add it.

Reply:
I guess we need to push a `space_id` field for index read view objects.

Reply:
Refactored `check_select_safety` to use an additional 'view' parameter.

Reply:
I guess we also need to add a read view id.

Reply:
Discussed with @locker: decided to refactor the logging part out of `check_select_safety` and add a `space` field to index read view tables storing the corresponding space read view table.